### PR TITLE
Update monitoring guidelines

### PR DESCRIPTION
**Pull request by sradco**
Signed-off-by: Shirly Radco <sradco@redhat.com>

**What this PR does / why we need it**:
Update monitoring guidelines with the required metrics prefixes, for each operator metrics.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/monitoring-guidelines.md [hunk 1]
```diff
@@ -9,7 +9,12 @@ The KubeVirt Users should have the same experience when searching for a node, co
 
 **Naming requirements:**
 1. Check if a similar Kubernetes metric, for node, container or pod, exists and try to align to it.
-2. KubeVirt metric for a running VM should have a `kubevirt_vmi_` prefix
+2. KubeVirt metrics prefixes:
+   1.  Running VM metrics should have a `kubevirt_vmi_` prefix
```
Comment (kwiesmueller) [inline docs/monitoring-guidelines.md:13]: If we do this, let's also already cover other components we have.
For example `api`. Our other components like `operator` could also have their respective prefix.
Tbh. I'm not entirely sure if metrics should really be prefixed with those details as the metric name should make sense without it and the metrics consumer should not necessarily care about the component emiting them, but the meaning they have.

File: docs/monitoring-guidelines.md [hunk 2]
```diff
@@ -9,7 +9,12 @@ The KubeVirt Users should have the same experience when searching for a node, co
...
+   2.  HCO operator metrics should have a `kubevirt_hco_` prefix
+   3.  Network operator metrics should have a `kubevirt_network_` prefix
```
Comment (phoracek) [inline docs/monitoring-guidelines.md:15]: Not sure if this one is fitting. Our network metrics that we can collect on kubevirt/kubevirt are related to VMI only. The other related network project we ship under HCO are not under the kubevirt/ GitHub organization. With this, we would not prefix metrics on e.g. kubernetes-nmstate with kubevirt_.

Comment (sradco) [reply docs/monitoring-guidelines.md:15]: This should be a prefix for network operator metrics that report on the operator itself. Not VMI network metrics.

Comment (phoracek) [reply docs/monitoring-guidelines.md:15]: Oh, I see. Looks good to me then, thanks for the clarification

File: docs/monitoring-guidelines.md (hunk 1 above)
Comment (sradco) [reply docs/monitoring-guidelines.md:13]: @kwiesmueller The intent of the prefix is to state for which operator it relates to. 
The operators metrics help with determining the operators health and performance.
Example: kubevirt_hco_out_of_band_modifications_count, which is specific to the hco operator.

Comment (kwiesmueller) [reply docs/monitoring-guidelines.md:13]: For metrics that exist across all our components I'd prefer them to have the same name and use different labels instead.
This makes them easier to query together.
Does that collide with this?

**Review by davidvossel — COMMENTED** [review]
I don't necessarily have any issues with the prefixes proposed here, however what is being proposed here exceeds the scope of what we can enforce from the kubevirt/kubevirt repo.

We can establish guidelines around what we control within the kubevirt/kubevirt repo here, however this document is now expanding that scope to include other projects as well, like hco, cdi, ssp, networking, etc...

I see two paths here. We can establish monitoring guidelines with each individual repo similar to what has been done with kubevirt, or we can establish a place for project wide guidelines like this. The latter would most likely belong in the community repo. It wouldn't be a "design proposal" but something new.

File: docs/monitoring-guidelines.md (hunk 1 above)
Comment (sradco) [reply docs/monitoring-guidelines.md:13]: @kwiesmueller No, it doesn't collide. They just need to have a kubevirt prefix.

**Pull request closed without merging**